2017 Nikola Samardzic <nikola@lnxsrv07.seas.ucla.edu>

* timespec.h (function timespec_cmp): changed to avoid integer overflow.

--- ./coreutils-with-bug/lib/timespec.h	2017-02-01 17:53:14.626005000 -0800
+++ ./newtimespec.h	2017-02-01 21:42:55.568543000 -0800
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec - b.tv_sec > 0)
+    return 1;
+  if(a.tv_sec - b.tv_sec < 0)
+    return -1;
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
